### ISSUE TITLE
initial conditions for Modelica.Stategraph.Examples.*

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#946]
**Reported by christian&#46;schubert on 28 Dec 2012 16:52 UTC**
While looking into the initial systems of the examples from Modelica.Stategraph I came to two questions:

1. 
The followinng lines are taken from Modelica.StateGraph.Interfaces.PartialTransition.
```
when enableFire then
  t_start = time;
end when;
```

Clearly, t_start is a discrete variable defined as
```
Modelica.SIunits.Time t_start;
```

Since there is neither an initial equation for t_start nor is it fixed=true.
I wonder if that initial system should be considered to be under determined?

2.
Also, we have
```
when outerStatePort.subgraphStatePort.suspend then
  oldActive = localActive;
end when;
```
in Modelica.StateGraph.Interfaces.PartialStep.

Since the evaluation of that when condition requires the knowledge of pre(outerStatePort.subgraphStatePort.suspend) I was wondering whether the initial system should provide that information which consequently would require an start value for the suspend variable. 

Should this model then be considered to be under determined? 

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/946

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 31 Dec 2012 12:57 UTC**
Replying to [ticket:946 christian.schubert@…]:

> While looking into the initial systems of the examples from Modelica.Stategraph I came to two questions:
> 
> 1. 
> The followinng lines are taken from Modelica.[wiki:StateGraph].Interfaces.[wiki:PartialTransition].
> ```
> when enableFire then
> t_start = time;
> end when;
> ```
> 
> Clearly, t_start is a discrete variable defined as
> ```
> Modelica.SIunits.Time t_start;
> ```
> 
> Since there is neither an initial equation for t_start nor is it fixed=true.
> I wonder if that initial system should be considered to be under determined?

Fixed in 5d2344829d9925e77c8e6dd03c0a8dbfe4cad8ea by providing the missing initial value.

Note, this is an uncritical bug, since the missing initial value does not influence the simulation result (the model is written in such a form, so that the value of t_start is only used, if set previously in the when-clause).

> 
> 2.
> Also, we have
> ```
> when outerStatePort.subgraphStatePort.suspend then
> oldActive = localActive;
> end when;
> ```
> in Modelica.[wiki:StateGraph].Interfaces.[wiki:PartialStep].
> 
> Since the evaluation of that when condition requires the knowledge of pre(outerStatePort.subgraphStatePort.suspend) I was wondering whether the initial system should provide that information which consequently would require an start value for the suspend variable. 
> 
> Should this model then be considered to be under determined?


This model is correct, since oldActive is initialized as:

```
initial equation 
  pre(oldActive) = pre(localActive);

```
Note, since variable oldActive is set in a when-clause, oldActive = pre(oldActive) if the when-clause is not active. Therefore, defining pre(oldActive) in the initial equation section, will also define a value for oldActive, when the when-clause is not active.

**Comment by modelica-trac-importer** [comment]
**Modified by otter on 1 Jan 2013 09:28 UTC**
